test/CMakeLists.txt

```diff
@@ -3,6 +3,17 @@
 #
 # Rahul, Jeff Ator, Ed Hartnett
 
+# This function builds, links, and runs a test program.
+function(create_test name kind)
+  add_executable(${name}_${kind} ${name}.F90)
+  target_link_libraries(${name}_${kind} PRIVATE bufr_${kind})
+  set_target_properties(${name}_${kind} PROPERTIES COMPILE_FLAGS "${fortran_${kind}_flags}")
+  set_target_properties(${name}_${kind} PROPERTIES Fortran_MODULE_DIRECTORY
+    ${CMAKE_CURRENT_BINARY_DIR}/include_${kind})
+  add_test(NAME ${name}_${kind} COMMAND ${name}_${kind})
+  target_compile_definitions(${name}_${kind} PUBLIC -DKIND_${kind})  
+endfunction()
+
 # Fetch test data from: https://ftp.emc.ncep.noaa.gov/static_files/public/bufr.tar
```

Comment:
Related to the below discussion about each test code being built and tested as _4, _8, and _d, shouldn't this flag to target_compile_definitions be `PRIVATE "${fortran_${kind}_defs}"` as it is in the IN tests section, rather than `PUBLIC -DKIND_${kind})`?  Or do these end up evaluating to the same thing?

In other words, what does `PUBLIC -DKIND_${kind})` end up doing here?

Reply:
Let me take a look and get back to you. I copied this from another NCEPLIBS...

Reply:
OK, looks like the only preprocessor flag set in existing tests is INTSIZE_8 for _8 builds.

There doesn't seem to be anything set for _d builds or _4 builds.

So what this does:
  target_compile_definitions(${name}_${kind} PUBLIC -DKIND_${kind})  

is for each test, defines either KIND_4, KIND_d, or KIND_8.

Reply:
Correct, there are no preprocessor flags for _d or _4 builds, because none are needed.

It's fine with me if you want to replace INTSIZE_8 with KIND_8 moving forward.  I presume the latter is just to keep things consistent with other libraries or with general industry practice?  Or just a personal preference?

Also, is the `set_target_properties(${name}_${kind} PROPERTIES Fortran_MODULE_DIRECTORY` line really needed here?  I've never had to use that before.  If it's really needed then fine, but it isn't used anywhere else for any other builds in this file, so if it's not really needed then my preference would be to keep things simpler and just remove it.

Reply:
OK, actually it seems that no preprocessor definition is required, since the variables can be declared integer 4 in the test and that will always work.

I have removed the fortran module directory line. This is not needed here because we are not using modules.